governance/program/src/error.rs

```diff
@@ -13,7 +13,7 @@ use thiserror::Error;
 pub enum GovernanceError {
     /// Invalid instruction passed to program
     #[error("Invalid instruction passed to program")]
-    InvalidInstruction,
+    InvalidInstruction = 500, // Start Governance custom errors from 500 to avoid conflicts with programs invoked via CPI
```

Comment:
We should really come up with a better custom error type that automatically adds the program id in some way

Reply:
Can't agree more. The fact that we can't surface correct errors from programs to users makes a very terrible UX. 

I looked into [ProgramError](https://github.com/solana-labs/solana/blob/master/sdk/program/src/program_error.rs) and in theory it should be just another error type `ProgramError:ProgramError(pubkey,u32`)  in addition to `ProgramError:Custom(u32)` but maybe I'm oversimplifying it without full understanding of the stack